### PR TITLE
Clean up IM training

### PR DESCRIPTION
**Pull request by jjc10**
Cleanup of how IMs are trained (no longer requiring a multi q wrapper) + implemented warm-up so that a classifier that has already converged on the validation set is no longer trained.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -160,4 +166,44 @@ def get_checkpoint(arch_name, dataset, exit_positions, bits):
...
+                                  drop_block_rate=None,
+                                  global_pool=None,
+                                  bn_tf=False,
+                                  bn_momentum=None,
+                                  bn_eps=None,
+                                  img_size=img_size)
+        if checkpoint is not None:
+            if 'state_dict' in checkpoint:
+                state_key = 'state_dict'
+            elif 'state_dict_ema' in checkpoint:
```
Comment (floregol) [inline src/utils.py:189]: do we still need this check for ema?

File: src/utils.py [hunk 2]
```diff
@@ -160,4 +166,44 @@ def get_checkpoint(arch_name, dataset, exit_positions, bits):
...
+                state_key = 'state_dict_ema'
+            else:
+                state_key = 'net'
+
+            # remove data parallel module wrapper
+            dict_without_data_parallel = {}
```
Comment (floregol) [inline src/utils.py:195]: and this? we can just scrap the parallel thing, or is it because it is stored by default in the checkpoint we are using?

**Review by floregol — APPROVED** [review]
